test/unit/node/app.test.ts

```diff
@@ -17,7 +17,7 @@ describe("createApp", () => {
   beforeAll(async () => {
     mockLogger()
 
-    const testName = "unlink-socket"
```

Comment:
Not sure if you're aware of this but there are [maximum unix socket path lengths of ~100 chars](https://unix.stackexchange.com/questions/367008/why-is-socket-path-length-limited-to-a-hundred-chars), so I had to shorten the test name to get this working on MacOS.

Without shortening it, there's some really wonky behavior where it thinks the socket already exists, which seems to be caused by the fact we're creating two sockets now.

Spent 2/3 hours debugging before I realized the socket path lengths were the issue. I'd suggest that we error out explicitly if the socket path lengths exceed the OS maximum, or at least shorten the test tmpdir paths, lest some poor developer come across this silent failure mode again...

Reply:
Oh wow that is wild.  Good to know!  Erroring out seems very wise to me.  Sounds like the perfect nightmare of a bug.